### DISCUSSION TITLE
wizard and relation

### DISCUSSION THREAD

**Discussion by photojo** [filamentphp/filament#6968] [Help] [answered]
I have Model user and model car and a relation between them.
So with Card & Repeater I can create a User AND and one or more cars in one form. Perfekt.

But can I also do so with a Wizard\Step::make...?

F.e. 
STEP 1: User
STEP 2: Car 1
STEP 3: Car 2
...
STEP X: finish

**Comment by photojo** [comment]
Anybody an idea? Is this possible at all?
Thank you so much in advance.

**Comment by pboivin** [comment]
Is the Wizard on a `CreateUser` page? What have you tried already? Can you share some code?

**Reply by photojo** [reply, marked as answer]
Hi pboivin!

I do not use CreateUser page, cause I do not work with the Adminpanel.

So, I use just filament tables.


 ```
protected function getTableQuery(): Builder
    {
        return Driver::query()->with(cars);
    }

```

```
 protected function getTableHeaderActions(): array
    {
        return [
            CreateAction::make('add')
            ->form([
             Wizard::make([
                        Wizard\Step::make('firstname')
                            ->schema([
                                TextInput::make('name'),
                            ]),
                        Wizard\Step::make('cars')
                            ->schema([
                               **TextInput::make('producer')**
                            ]),

                    ]),

```

 **TextInput::make('producer')** this should be inserted in Model Car, which is related with Model Driver

I hope, you understand my problem.



By the way: when I use **EditAction** with Card & Repeater like
```
 Repeater::make('cars')->label('Producer')
                                        ->relationship()
```

But I want add new rows with a wizard, step by step and write them in **two tables/models**.

Thanks a lot for spending time one my problem.

**Comment by pboivin** [comment]
Hey @photojo. So if I understand correctly, instead of creating the related cars with a Repeater, you want to have each car as a Wizard Step? I don't know if there's an easy way to do that, but I can think of a hard way 😅️

Using a custom action, you can access all the data from the steps when the form is submitted:

```php
    Action::make('add')
        ->form([
            Wizard::make([
                // ...
            ]),
        ])
        ->action(function ($data): void {
            dd($data);
        });
```

You may be able to manually create the related cars from there.

**Comment by photojo** [comment]
Hey @pboivin. That works really fine. Thank you so much!
Greetings
Jo
